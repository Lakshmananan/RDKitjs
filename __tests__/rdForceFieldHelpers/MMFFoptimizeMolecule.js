'use strict';

let RDKit;
beforeEach(async () => (RDKit = await require('../..')));

test('default parameters', () => {
  const smiles = 'COCO';
  const mol = RDKit.Mol.fromSmiles(smiles);
  mol.addHs();
  mol.embedMolecule();
  const result = mol.MMFFoptimizeMolecule();
  expect(result).toHaveLength(2);
  expect(result[0]).toBe(0);
  expect(result[1]).toBeCloseTo(-19.4, 1);
  expect(mol.toMolBlock()).toMatchSnapshot();
});
