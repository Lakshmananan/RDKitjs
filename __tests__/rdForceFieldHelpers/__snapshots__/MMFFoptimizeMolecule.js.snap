// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`default parameters 1`] = `
"
     RDKit          3D

 10  9  0  0  0  0  0  0  0  0999 V2000
   -1.1845    0.1513    0.5226 C   0  0  0  0  0  0  0  0  0  0  0  0
   -0.5788   -0.6520   -0.4770 O   0  0  0  0  0  0  0  0  0  0  0  0
    0.8339   -0.5462   -0.4849 C   0  0  0  0  0  0  0  0  0  0  0  0
    1.2588    0.6965   -1.0083 O   0  0  0  0  0  0  0  0  0  0  0  0
   -2.2647   -0.0151    0.4897 H   0  0  0  0  0  0  0  0  0  0  0  0
   -0.9935    1.2124    0.3370 H   0  0  0  0  0  0  0  0  0  0  0  0
   -0.8204   -0.1290    1.5158 H   0  0  0  0  0  0  0  0  0  0  0  0
    1.2564   -0.7071    0.5135 H   0  0  0  0  0  0  0  0  0  0  0  0
    1.2198   -1.3304   -1.1423 H   0  0  0  0  0  0  0  0  0  0  0  0
    1.2730    1.3195   -0.2661 H   0  0  0  0  0  0  0  0  0  0  0  0
  1  2  1  0
  2  3  1  0
  3  4  1  0
  1  5  1  0
  1  6  1  0
  1  7  1  0
  3  8  1  0
  3  9  1  0
  4 10  1  0
M  END
"
`;
